### PR TITLE
Catalogue not displayed on nightly

### LINKED ISSUE #1214: Catalog is not displayed in nightly
[Catalog view](https://nightly.apinf.io/catalogue) in nightly displays only the top bar without any APIs

# Steps to reproduce

1. Log into https://nightly.apinf.io
2. Click Catalogue
3. (Optionally, try clicking any of the buttons in the top bar: My APIs, My Bookmarks or toggling the views)

# Observed behavior
The view does not contain any APIs, i.e. there is no catalog

# Expected behavior
The default API catalog view (card view) is displayed

# Environment

- Web browser: Chrome 51.0.2704.103
- Operating system: Windows 10

### PR DESCRIPTION
**Pull request by jykae**
Closes #1214 

## Proposed changes
- check bookmarks result, user might not have bookmarks, which results undefined.
- add reactivevar import
- paranoid check user query, in case of undefined

### CODE REVIEW

**Comment by jykae** [comment]
Tried deploying this branch for temporary server to test in production one exception occurring on nightly but not in development.
However I could not catch this error on development and not in test deployment,
```
TypeError: Cannot read property 'splice' of null
    at https://nightly.apinf.io/2a2fb884adc53dffc6e76ae838b2444f03baafbd.js?meteor_js_resource=true:107:29128
    at Object.o.nonreactive (https://nightly.apinf.io/2a2fb884adc53dffc6e76ae838b2444f03baafbd.js?meteor_js_resource=true:52:4311)
    at Object.removedAt (https://nightly.apinf.io/2a2fb884adc53dffc6e76ae838b2444f03baafbd.js?meteor_js_resource=true:107:28856)
    at Object.removed (https://nightly.apinf.io/2a2fb884adc53dffc6e76ae838b2444f03baafbd.js?meteor_js_resource=true:103:1831)
    at https://nightly.apinf.io/2a2fb884adc53dffc6e76ae838b2444f03baafbd.js?meteor_js_resource=true:70:1323
    at Array.forEach (native)
    at Function.A.each.A.forEach (https://nightly.apinf.io/2a2fb884adc53dffc6e76ae838b2444f03baafbd.js?meteor_js_resource=true:1:865)
    at r.diffQueryOrderedChanges (https://nightly.apinf.io/2a2fb884adc53dffc6e76ae838b2444f03baafbd.js?meteor_js_resource=true:70:1281)
    at l (https://nightly.apinf.io/2a2fb884adc53dffc6e76ae838b2444f03baafbd.js?meteor_js_resource=true:103:1417)
    at https://nightly.apinf.io/2a2fb884adc53dffc6e76ae838b2444f03baafbd.js?meteor_js_resource=true:103:802
```
Catalogue and each tab opens ok for me in development and test production deployment with these changes. Could someone of @apinf/developers do review?

**Comment by nnn** [comment]
Getting:

```
2a2fb88….js?meteor_js_resource=true:3 Exception from Tracker recompute function:
2a2fb88….js?meteor_js_resource=true:3 TypeError: Cannot read property 'username' of undefined
    at https://nightly.apinf.io/2a2fb884adc53dffc6e76ae838b2444f03baafbd.js?meteor_js_resource=true:428:4436
    at Array.map (native)
    at Function.A.map.A.collect (https://nightly.apinf.io/2a2fb884adc53dffc6e76ae838b2444f03baafbd.js?meteor_js_resource=true:1:1117)
    at t.e [as getApiManagersByName] (https://nightly.apinf.io/2a2fb884adc53dffc6e76ae838b2444f03baafbd.js?meteor_js_resource=true:428:4386)
    at https://nightly.apinf.io/2a2fb884adc53dffc6e76ae838b2444f03baafbd.js?meteor_js_resource=true:126:2518
    at s.call (https://nightly.apinf.io/2a2fb884adc53dffc6e76ae838b2444f03baafbd.js?meteor_js_resource=true:126:1929)
    at s.mustacheImpl (https://nightly.apinf.io/2a2fb884adc53dffc6e76ae838b2444f03baafbd.js?meteor_js_resource=true:126:1184)
    at Object.s.mustache (https://nightly.apinf.io/2a2fb884adc53dffc6e76ae838b2444f03baafbd.js?meteor_js_resource=true:126:1243)
    at ._render (https://nightly.apinf.io/2a2fb884adc53dffc6e76ae838b2444f03baafbd.js?meteor_js_resource=true:440:29692)
    at l (https://nightly.apinf.io/2a2fb884adc53dffc6e76ae838b2444f03baafbd.js?meteor_js_resource=true:107:20600)

**Comment by nnn** [comment]
@jykae, if it's working on test deployment, let's merge it and do I manual deployment to nightly, since it's a catalogue is a big thing.

**Comment by jykae** [comment]
@nnn Ok, I think "username" is now checked as much as one can. I could manual deploy on nightly if you merge this.

**Comment by jykae** [comment]
@nnn Deployed this branch on nightly, https://travis-ci.org/apinf/api-umbrella-dashboard/builds/140729834 

Check https://nightly.apinf.io/catalogue

**Comment by jykae** [comment]
As you notice there's one weird edge case, where this "if-statement" does not happen, https://github.com/apinf/api-umbrella-dashboard/blob/bfa7d9475fc0ea0f7bb02cb97401bff5db91059a/apis/collection/helpers.js#L129

@nnn Is it ok to return nothing like it is now, or should we return there also "admin"? What you think?

**Comment by nnn** [comment]
@jykae, sounds right for me. @brylie, what do you think?

File: catalogue/server/publications.js [hunk 1]
```diff
@@ -10,11 +10,16 @@ Meteor.publish('catalogue', function (options) {
...
-    const bookmarkedApiIds = userBookmarks.apiIds;
+    
+    // Check that result is not undefined
+    if( userBookmarks ) {
+      // Get bookmarked API IDs
+      bookmarkedApiIds = userBookmarks.apiIds;
+    }
 
     // Set up query object to contain bookmarked API IDs
     query = {_id: {$in: bookmarkedApiIds}};
```
Comment (brylie) [inline catalogue/server/publications.js]: This should also be moved inside of the `if (userBookmarks) { .. }` block. 

Also, wait until @nnn merges changes in pr #1221, as some of these changes overlap.

**Comment by brylie** [comment]
This PR looks good. Lets hold off until we get PR #1221 merged, and see how that relates to this PR. I.e. the changes to `publication.js` may not be necessary after the PR #1221 merge.

**Pull request merged**